File: queue/queue-class/queue.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
    
class Queue:
    def __init__(self):
        self.front = self.rear = None
        self.size = 0
    
    def enqueue(self, data):
        new_node = Node(data)
        if not self.front:
            self.front = self.rear = new_node
        else:
            self.rear.next = new_node
            self.rear = new_node
        self.size += 1
    
    def dequeue(self):
        if not self.front:
            return None
        data = self.front.data
        self.front = self.front.next

        # rear could also hold the last element, so we need to check
        # if front becomes None, then rear becomes None
        if not self.front:
            self.rear = None
        
        self.size -= 1
        return data
    
    def peek(self):
        if not self.front:
            return None
        
        return self.front.data
    
    def get_size(self):
        return self.size
    
    def isEmpty(self):
        return self.size == 0
    
if __name__ == "__main__":
    queue = Queue();
    
    queue.enqueue(1)
    queue.enqueue(2)
    queue.enqueue(3)
    # Display front element
    print("Front element:", queue.peek())
    # Dequeue and display the dequeued element
    print("Dequeued:", queue.dequeue())
    # Display front element again
    print("Front element:", queue.peek())
    # Display the size of the queue
    print("Queue size:", queue.get_size())

File: Queue/Simple Queue/queue.java
class Queue<T> {
  private Node<T> front, rear;
  private int size;

  // Node class for storing data and the reference to the next node
  private static class Node<T> {
      T data;
      Node<T> next;

      Node(T data) {
          this.data = data;
          this.next = null;
      }
  }

  public Queue() {
      front = rear = null;
      size = 0;
  }

  // Add an element to the queue
  public void enqueue(T data) {
      Node<T> newNode = new Node<>(data);
      if (rear == null) {
          front = rear = newNode;
      } else {
          rear.next = newNode;
          rear = newNode;
      }
      size++;
  }

  // Remove an element from the queue
  public T dequeue() {
      if (front == null) {
          return null;
      }
      T data = front.data;
      front = front.next;
      if (front == null) {
          rear = null;
      }
      size--;
      return data;
  }

  // Check the front element of the queue
  public T peek() {
      if (front == null) {
          return null;
      }
      return front.data;
  }

  // Check if the queue is empty
  public boolean isEmpty() {
      return size == 0;
  }

  // Get the size of the queue
  public int size() {
      return size;
  }
}

// Example usage
public class Solution {
  public static void main(String[] args) {
      Queue<Integer> queue = new Queue<>();

      queue.enqueue(1);
      queue.enqueue(2);
      queue.enqueue(3);

      System.out.println("Front element: " + queue.peek());
      System.out.println("Dequeued: " + queue.dequeue());
      System.out.println("Front element: " + queue.peek());
      System.out.println("Queue size: " + queue.size());
  }
}